homeassistant/components/filesize/sensor.py

```diff
@@ -137,9 +137,8 @@ async def _async_update_data(self) -> dict[str, float | int | datetime]:
             raise UpdateFailed(f"Can not retrieve file statistics {error}") from error
 
         size = statinfo.st_size
-        last_updated = datetime.fromtimestamp(statinfo.st_mtime).replace(
-            tzinfo=dt_util.UTC
-        )
+        last_updated = dt_util.as_local(datetime.fromtimestamp(statinfo.st_mtime))
```

Comment:
```suggestion
        last_updated = datetime.utcfromtimestamp(statinfo.st_mtime)
```

Reply:
That would be a more correct way of handling indeed 👍 
Thanks!

Reply:
Unfortunately it doesn't work because the raw value of `os.stat` is localized.

Reply:
The raw value from `os.stat` is a [POSIX timestamp](https://en.wikipedia.org/wiki/Unix_time) which is the same regardless of the system timezone setting.  

```
>>> os.stat('/etc/passwd')
os.stat_result(st_mode=33188, st_ino=159832373, st_dev=16777234, st_nlink=1, st_uid=0, st_gid=0, st_size=7868, st_atime=1648771486, st_mtime=1648279273, st_ctime=1648771486)
>>> datetime.datetime.fromtimestamp(1648279273)
datetime.datetime(2022, 3, 25, 21, 21, 13)
>>> datetime.datetime.utcfromtimestamp(1648279273)
datetime.datetime(2022, 3, 26, 7, 21, 13)
>>> 
bdraco@MacBook-Pro-4 components % ls -l /etc/passwd
-rw-r--r--  1 root  wheel  7868 Mar 25 21:21 /etc/passwd
```

Reply:
I have a different result here on both prod and dev container :-S

Reply:
You should get a different result for `fromtimestamp` since it will be in your local time. You should get the same result for `utcfromtimestamp` though

Reply:
Are you saying you get a different result for the following based on your system's timezone setting?
```
>>> import os
>>> os.stat('/etc/passwd')
os.stat_result(st_mode=33188, st_ino=33786339, st_dev=201, st_nlink=1, st_uid=0, st_gid=0, st_size=1336, st_atime=1650110744, st_mtime=1625535202, st_ctime=1625535202)
>>> 
```

Reply:
```
~# python
Python 2.7.18 (tags/DSM6-2-25482-200701:af4aa2a, Mar  4 2021, 18:10:31) 
[GCC 4.9.3 20150311 (prerelease)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import os
>>> os.stat('/volume1/docker/89-homeassistant/configuration.yaml')
posix.stat_result(st_mode=33261, st_ino=125764545, st_dev=64512, st_nlink=1, st_uid=1026, st_gid=100, st_size=14364, st_atime=1650180749, st_mtime=1650180739, st_ctime=1650180739)
>>> 
root@mauinasa:~# docker exec -it 89-homeassistant /bin/bash 
bash-5.1# python3
Python 3.9.9 (main, Feb  8 2022, 09:16:53) 
[GCC 10.3.1 20210424] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import os
>>> os.stat('/config/configuration.yaml')
os.stat_result(st_mode=33261, st_ino=125764545, st_dev=64512, st_nlink=1, st_uid=1026, st_gid=100, st_size=14364, st_atime=1650180749, st_mtime=1650180739, st_ctime=1650180739)

```

Reply:
Started dev container from scratch and now it works :confused: